public/app/app.ts

```diff
@@ -225,7 +225,14 @@ export class GrafanaApp {
         await preloadPlugins(awaitedAppPlugins, extensionsRegistry, 'frontend_awaited_plugins_preload');
       }
 
-      setPluginExtensionGetter(createPluginExtensionsGetter(extensionsRegistry));
+      // Accessing this as a singleton will create some problems to URL handling for sidecar. Right now it's ok
+      // because we use the URl just to get some pseudo app IDs for core functionality.
+      setPluginExtensionGetter(
+        createPluginExtensionsGetter(
+          extensionsRegistry,
+          getSidecarHelpers(store.dispatch, () => store.getState().appSidecar.appId, locationService)
+        )
+      );
```

Comment:
Hmm, using the redux store directly means that we need to "push down" these helpers to where they will be actually used (which adds some complexity), and also means that the core extensions code is now depending on something more specific: the sidecar slice.

What's the reason against using the event bus for controlling the sidecar for example? (Under the hood the state could still be managed in the existing redux slice.) In that case the core extensions code could be simpler, and we also wouldn't need to pass the helpers down from the `app.ts`.

Reply:
The problem with using event bus for this is again the singleton nature of this. Like at the same level these functions could just be exposed through grafana/runtime so you could import it anywhere. The problem is that they would be unscoped and so would need to be like `openApp(appID)` which means you could supply any appID there so you could close app A from app B, which may not be a huge deal but I don't think it should be allowed. So we have to somehow scope those functions so they have the appID already injected and so you can open and close only your own app from your extension.

Events would have the same problem the even would have to have a payload with the app ID or we would have to have a scoped even bus which than would need to be propagated creating the same issue.

Reply:
Based on our discussion last week I think we (plugins-platform) would still like this communication to be done through the event bus.

**Known risks and downsides**
- _the EventBus is a singleton_ and in case we would like to provide different event bus services for the different parts of the application in the future, then changing this might have difficulties. 
- _events can currently be sent by whoever has access to the main EventBus_, which involves plugins, too. Technically this means that if we go down this path, then any plugin can control (open/close) the sidecar views. Maybe this is not ideal, however we think it's acceptable, mostly because we have a control over what plugins can do in our catalog (via code reviews and automated checks). On this note - couldn't extensions that have access to a `openApp()` function (bound to their plugin id) also close the currently open sidecar view when they call it?

**Benefits**
- _The implementation remains simpler._  The sidecar helpers wouldn't need to be passed down from the top (`createPluginExtensionsGetter()` -> `getPluginExtensions()` -> `getLinkExtensionOnClick()`), and the helpers for extensions could live in the same place (somewhere in the extensions utils). Also the extensions framework could remain a more general functionality [that doesn't depend on anything "specific"](https://github.com/grafana/grafana/pull/91648/files#diff-f95e2a47d4ba61bb2a27c422b0dd3b25e05e93444d0339abc29f2ec36c469a94R14). This state would help us to make [the refactoring of the plugin extensions core codebase](https://github.com/grafana/grafana/issues/89473) simpler.

(This doesn't mean that we cannot inject services as dependencies for the extensions framework later instead of using singletons.)

### Ideas
We would probably keep how the [sidecar helpers are bound to the pluginId now](https://github.com/grafana/grafana/pull/91648/files#diff-f95e2a47d4ba61bb2a27c422b0dd3b25e05e93444d0339abc29f2ec36c469a94R91), but the actual helper function could come from the utils as a static import, and just publish an event on the event bus internally (e.g. `export class OpenAppInSideviewEvent extends BusEventWithPayload<{ pluginId: string }>`), A sidecar specific service then could listen to these events and dispatch the necessary actions for the reducer. For the new extension helper functions we can probably add an [`@experimental` TSDoc](https://tsdoc.org/pages/tags/experimental/), too.

Please let us know @aocenas if you would need any help with this.

Reply:
> On this note - couldn't extensions that have access to a openApp() function (bound to their plugin id) also close the currently open sidecar view when they call it?

Yes the difference is whether it can open and close any other pluginID.

> actual helper function could come from the utils as a static import, and just publish an event on the event bus internally (e.g. export class OpenAppInSideviewEvent extends BusEventWithPayload<{ pluginId: string }>)

Not sure how this would be technically possible while keeping the pluginID bound to the function.

Reply:
Will do the changes here to internally use the even bus that does not fundamentally change this PR.

Reply:
So I made a commit here https://github.com/grafana/grafana/pull/91648/commits/04ed160127e07201d8870c09382bab40c2a143d9.

Honestly though not sure what we gained with that don't think the code is better with that change.

Reply:
Resolving this one due to https://github.com/grafana/grafana/pull/91648/commits/53c69085abab046679c47d923c59740ac53e26b6.